test/dlopen/test_dlopen.c

```diff
@@ -82,6 +84,18 @@ int main()
       fprintf(stderr, "ERROR: call to return_five in libnum.so was not wrapped by correct_return_five\n");
       had_error = -1;
    }
+    /* Test 3: Does the dlsym implementation find the second occurrence of the symbol */
+    test_retfive = (int (*)(void)) dlsym(RTLD_NEXT, "return_five");
+    if (test_retfive != NULL) {
+        fprintf(stderr, "ERROR: call to return_five should not be found in RTLD_NEXT\n");
+        had_error = -1;
+    }
+    /* Test 4: Does the dlsym implementation find the first occurrence of the symbol */
+    test_retfive = (int (*)(void)) dlsym(RTLD_DEFAULT, "return_five");
+    if (test_retfive == NULL || test_retfive() !=5) {
+        fprintf(stderr, "ERROR: call to return_five should be found in RTLD_DEFAULT\n");
+        had_error = -1;
+    }
```

Comment:
We need a more substantive test case, one where there's multiple symbols that RTLD_NEXT could return, and we know that our RTLD_NEXT re-implementation is doing the right thing. I've already written that, let me find it and email it.

Reply:
I added a second library called libnum2. Where I am now checking the next symbol.